Update docs for develop (#627)
kubernetes-sigs/kwok: site/content/en/docs/contributing/development.md

  - kwokctl - Main entry point for `kwokctl`
- pkg
  - apis - API definitions
    - config
      - v1alpha1 - Configuration API definitions for parsing and converting only
    - internalversion - For all internal use only
    - v1alpha1 - API definitions for parsing and converting only
  - config - Configuration utilities
  - kwok - `kwok` implementation
  - kwokctl - `kwokctl` implementation

## Build and Run

> ### Start with Container

Reviewer: ```suggestion
### Start with Containers
```

Build `kwok` image and `kwokctl` binary.

```bash
IMAGE_PREFIX=localhost BUILDER=docker make build build-image
```

On a successful build, the binaries will be located in `./bin/$(go env GOOS)/$(go env GOARCH)`, the image will be tagged as `localhost/kwok:${tag}` and can be found in `docker images`.

Now, we can create cluster using `kwokctl` with `docker` runtime.

``` bash
./bin/$(go env GOOS)/$(go env GOARCH)/kwokctl create cluster \
  --runtime=docker
```

By the way, you can also use `podman` or `nerdctl` as the builder and the runtime.

> ### Start without Container

Reviewer: ```suggestion
### Start with platform-specific binaries
```

Build `kwok` and `kwokctl` binaries.

``` bash
make build
```

On a successful build, the binaries will be located in `./bin/$(go env GOOS)/$(go env GOARCH)`.

> Note that if running in Non-Linux then you will need take look [Without Container] to build the Kubernetes binaries.

Reviewer: ```suggestion
Note that if running in Non-Linux platforms, then you will need to follow [Without Container] to build the Kubernetes binaries locally.
```
